Java/src/main/java/net/hypixel/api/reply/PlayerReply.java

```diff
@@ -1,23 +1,358 @@
+         */
+        public long getNetworkExp() {
+            return getNumberProperty("networkExp", 0).longValue();
+        }
+
+        /**
+         * @return The player's precise network level (includng progress)
+         */
+        public double getNetworkLevel() {
+            return ILeveling.getExactLevel(getNetworkExp());
+        }
+
+        /**
+         * @return The total amount of karma points earned by this player
+         */
+        public long getKarma() {
+            return getNumberProperty("karma", 0).longValue();
+        }
+
+        /**
+         * @return The date when the player first connected to Hypixel
+         */
+        public Date getFirstLoginDate() {
+            return new Date(getLongProperty("firstLogin", 0));
+        }
+
+        /**
+         * @return The date when the player most recently connected to Hypixel
+         */
+        public Date getLastLoginDate() {
+            return new Date(getLongProperty("lastLogin", 0));
+        }
+
+        /**
+         * @return The date when the player more recently disconnected from Hypixel
+         */
+        public Date getLastLogoutDate() {
+            return new Date(getLongProperty("lastLogout", 0));
+        }
+
+        /**
+         * @return The date when the player first connected to Hypixel
+         * @see net.hypixel.api.HypixelAPI#getStatus(UUID)
+         * @deprecated The status endpoint is recommended for checking a player's online status
+         */
+        @Deprecated
+        public boolean isOnline() {
+            return getLongProperty("lastLogin", 0) > getLongProperty("lastLogout", 0);
+        }
+
+        /**
+         * The highest network rank that this player has; prefixes are not taken into consideration
+         * <p>
+         * Example: If... <ul>
+         * <li>A player's base rank is MVP+</li>
+         * <li>They have a subscription for MVP++</li>
+         * <li>They are a staff member with the HELPER rank</li>
+         * </ul>
+         * ...then this method will return "HELPER".
+         *
+         * @return This player's highest network rank, or "NONE" if they do not have any
+         * @see <a href=https://github.com/HypixelDev/PublicAPI/wiki/Common-Questions#how-do-i-get-a-players-rank-prefix>"How
+         * do I get a player's rank prefix?"</a>
+         */
+        public String getHighestRank() {
+            if (hasRankInField("rank")) {
+                return getStringProperty("rank", DEFAULT_RANK);
+
+            } else if (hasRankInField("monthlyPackageRank")) {
+                return getStringProperty("monthlyPackageRank", DEFAULT_RANK);
+
+            } else if (hasRankInField("newPackageRank")) {
+                return getStringProperty("newPackageRank", DEFAULT_RANK);
+
+            } else if (hasRankInField("packageRank")) {
+                return getStringProperty("packageRank", DEFAULT_RANK);
+            }
+
+            return DEFAULT_RANK;
```

Comment:
rank can be NORMAL, but can't be NONE; package ranks can be NONE, but can't be NORMAL. Knowing that it could be optimized a bit, but that's not important

Reply:
In [the screenshot](https://staticassets.hypixel.net/news/5df2c1e8abfe5.comparison.png) posted on their private API blog post, to determine if a player has a rank, they check each rank-related field (minus monthlyPackageRank, which wasn't needed) against both `NONE` and `NORMAL`. I feel like it's better to be safe in this case.